sdk/src/ingredient.rs

```diff
@@ -649,17 +660,45 @@ impl std::fmt::Display for Ingredient {
     }
 }
 
-#[derive(Default)]
-/// This defines optional actions when creating [`Ingredient`]s from files.
-pub struct IngredientOptions {
-    /// This allows setting the title for the ingredient. (If `None`, then the default behavior is to use the file's name.)
-    pub title: Option<&'static str>,
+/// This defines optional operations when creating [`Ingredient`] structs from files.
+#[cfg(feature = "file_io")]
+pub trait IngredientOptions {
+    /// This allows setting the title for the ingredient.
+    ///
+    /// If it returns `None`, then the default behavior is to use the file's name.
+    fn title(&self, _path: &Path) -> Option<String> {
+        None
+    }
 
-    /// If `true`, then generate a Blake3 hash over the source asset and store it in the ingredient.
+    /// Returns an optional hash value for the ingredient
+    ///
     /// This can be used to test for duplicate ingredients or if a source file has changed.
-    pub make_hash: bool,
+    /// If hash is_some() Manifest.add_ingredient will dedup matching hashes
+    fn hash(&self, _path: &Path) -> Option<String> {
+        None
+    }
+
+    /// Returns an optional thumbnail image representing the asset
+    ///
+    /// The first value is the content type of the thumbnail, i.e. image/jpeg
+    /// The second value is bytes of the thumbnail image
+    /// The default is to have no thumbnail, so you must provide an override to have a thumbnail image
+    fn thumbnail(&self, _path: &Path) -> Option<(String, Vec<u8>)> {
+        #[cfg(feature = "add_thumbnails")]
```

Comment:
This can now be written as:
if cfg!(feature  = "add_thumbnails") {
do something
} else {
do something else 
}

Reply:
Ah is that new? I didn't know that was an option. thanks.

Reply:
updated to use this syntax